cloud/openstack/os_router_facts.py

```diff
@@ -0,0 +1,141 @@
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this software.  If not, see <http://www.gnu.org/licenses/>.
+
+try:
+    import shade
+    HAS_SHADE = True
+except ImportError:
+    HAS_SHADE = False
+
+DOCUMENTATION = '''
+---
+module: os_router_facts
+short_description: Retrieve facts about one or more OpenStack routers.
+version_added: "2.2"
+author: "Greg Hellings (@greg-hellings)"
+description:
+    - Retrieve facts about one or more routers from OpenStack.
+requirements:
+    - "python >= 2.6"
+    - "shade"
+options:
+   router:
+     description:
+        - Name or ID of the Router
+     required: false
+   filters:
+     description:
+        - A dictionary of meta data to use for further filtering.  Elements of
+          this dictionary may be additional dictionaries.
+     required: false
+extends_documentation_fragment: openstack
+'''
+
+EXAMPLES = '''
+# Gather facts about previously created routers
+- os_router_facts:
+    auth:
+      auth_url: https://your_api_url.com:9000/v2.0
+      username: user
+      password: password
+      project_name: someproject
+- debug: var=openstack_routers
+
+# Gather facts about a previously created router by name
+- os_router_facts:
+    auth:
+      auth_url: https://your_api_url.com:9000/v2.0
+      username: user
+      password: password
+      project_name: someproject
+    name:  router1
+- debug: var=openstack_routers
+
+# Gather facts about a previously created router with filter (note: name and
+  filters parameters are Not mutually exclusive)
+- os_router_facts:
+    auth:
+      auth_url: https://your_api_url.com:9000/v2.0
+      username: user
+      password: password
+      project_name: someproject
+    filters:
+      tenant_id: 55e2ce24b2a245b09f181bf025724cbe
+      subnets:
+        - 057d4bdf-6d4d-4728-bb0f-5ac45a6f7400
+        - 443d4dc0-91d4-4998-b21c-357d10433483
+- debug: var=openstack_routers
+'''
+
+RETURN = '''
+openstack_routers:
+    description: has all the openstack facts about the routers
+    returned: always, but can be null
+    type: complex
+    contains:
+        id:
+            description: Unique UUID.
+            returned: success
+            type: string
+        name:
+            description: Name given to the network.
+            returned: success
+            type: string
+        status:
+            description: Network status.
+            returned: success
+            type: string
+        subnets:
+            description: Subnet(s) included in this network.
+            returned: success
+            type: list of strings
+        tenant_id:
+            description: Tenant id associated with this network.
+            returned: success
+            type: string
+        shared:
+            description: Network shared flag.
+            returned: success
+            type: boolean
+'''
```

Comment:
These return values are not valid as you simply copied/pasted from the network facts module.